### PR TITLE
feat: add warning styles to Button (Primary, Secondary, Default, SystemUI)

### PR DESCRIPTION
**Pull request by ykelkar-eightfold**
## SUMMARY:
Add warning styles to Button (Primary, Secondary, Default, SystemUI)

## JIRA TASK (Eightfold Employees Only):
https://eightfoldai.atlassian.net/browse/ENG-73493

## CHANGE TYPE:

- [ ] Bugfix Pull Request
- [x] Feature Pull Request

## TEST COVERAGE:

- [x] Tests for this change already exist
- [ ] I have added unittests for this change

## TEST PLAN:
Please test locally with `warning` prop set to `true` for Primary, Secondary, Default, and SystemUI Buttons.

See the styles below when setting the warning prop to true for the buttons:

https://github.com/EightfoldAI/octuple/assets/102243974/f892532d-bedb-4259-aeb7-02f1af9bef0b

### CODE REVIEW

File: src/components/Button/button.module.scss [hunk 1]
```diff
@@ -871,6 +976,110 @@
...
+    background: transparent;
+  }
+
+  &:not(.transparent) {
+    background: var(--bg);
+  }
+
+  &:not(.transparent):not([disabled]):hover {
+    --bg: var(--button-system-ui-disruptive-hover-background-color);
+    --color: var(--button-system-ui-wardisruptivening-hover-text-color);
```
Comment (dkilgore-eightfold) [inline src/components/Button/button.module.scss:1074]: Typo

File: src/styles/themes/_default-theme.scss [hunk 2]
```diff
@@ -313,6 +313,10 @@
   --warning-hover-color: #ce811a;
   --warning-hover-background-color: #ffe3b0;
   --warning-hover-border-color: #c97e19;
+  --warning-background1-color: var(--orange-background1-color); // Surface 1
+  --warning-background2-color: var(--orange-background2-color); // Surface 2
+  --warning-background3-color: var(--orange-background3-color); // Surface 3
+  --warning-secondary-color: #c97e19;
```
Comment (dkilgore-eightfold) [inline src/styles/themes/_default-theme.scss:319]: Does this not exist? If not, use an existing color.

File: src/styles/themes/_default-theme.scss [hunk 3]
```diff
@@ -790,7 +870,7 @@
     --button-secondary-disruptive-border-style
   );
   --button-secondary-disruptive-hover-border-color: var(
-    --button-secondary-disruptive-hover-background-color
+    --button-secondary-disruptive-border-color
```
Comment (dkilgore-eightfold) [inline src/styles/themes/_default-theme.scss:873]: revert this, as these button colors are to spec.

File: src/styles/themes/_default-theme.scss [hunk 4]
```diff
@@ -805,13 +885,52 @@
     --button-secondary-disruptive-border-style
   );
   --button-secondary-disruptive-active-border-color: var(
-    --button-secondary-disruptive-active-background-color
+    --button-secondary-disruptive-border-color
```
Comment (dkilgore-eightfold) [inline src/styles/themes/_default-theme.scss:888]: Revert.

File: src/styles/themes/_default-theme.scss [hunk 5]
```diff
@@ -805,13 +885,52 @@
...
+    --button-secondary-warning-border-width
+  );
+  --button-secondary-warning-active-border-style: var(
+    --button-secondary-warning-border-style
+  );
+  --button-secondary-warning-active-border-color: var(
+    --button-secondary-warning-border-color
+  );
+  --button-secondary-disruptive-split-divider-color: var(--warning-color);
+  // ------ Secondary Disruptive Button theme ------
```
Comment (dkilgore-eightfold) [inline src/styles/themes/_default-theme.scss:932]: Typo.

File: src/styles/themes/_default-theme.scss [hunk 6]
```diff
@@ -805,13 +885,52 @@
...
+  --button-secondary-warning-active-border-width: var(
+    --button-secondary-warning-border-width
+  );
+  --button-secondary-warning-active-border-style: var(
+    --button-secondary-warning-border-style
+  );
+  --button-secondary-warning-active-border-color: var(
+    --button-secondary-warning-border-color
+  );
+  --button-secondary-disruptive-split-divider-color: var(--warning-color);
```
Comment (dkilgore-eightfold) [inline src/styles/themes/_default-theme.scss:931]: Typo.

File: src/components/Button/Button.tsx [hunk 7]
```diff
@@ -122,17 +123,37 @@ export const Button: FC<ButtonProps> = React.forwardRef(
...
+      {
+        [styles.buttonSystemUiDisruptive]:
+          disruptive && mergedVariant === ButtonVariant.SystemUI,
+      },
+      {
+        [styles.buttonDefaultWarning]:
+          warning && mergedVariant === ButtonVariant.Default,
+      },
+      {
+        [styles.buttonDefaultDisruptive]:
```
Comment (dkilgore-eightfold) [inline src/components/Button/Button.tsx:138]: Add UTs snapshotting the warning state for each button type.

File: src/components/Button/Button.types.ts [hunk 8]
```diff
@@ -225,4 +225,9 @@ export interface ButtonProps extends NativeButtonProps {
    * @default ButtonVariant.Default
    */
   variant?: ButtonVariant;
+  /**
+   * The button warning state.
+   * @default false
+   */
+  warning?: boolean;
```
Comment (dkilgore-eightfold) [inline src/components/Button/Button.types.ts:232]: Alphebetize props.

File: src/components/Button/button.module.scss [hunk 9]
```diff
@@ -755,7 +825,7 @@
   }
 }
 
-.button-disruptive {
+.button-default-disruptive {
```
Comment (dkilgore-eightfold) [inline src/components/Button/button.module.scss:828]: revert.

**Review by dkilgore-eightfold — CHANGES_REQUESTED** [review]

File: src/styles/themes/_default-theme.scss (hunk 2 above)
Comment (dkilgore-eightfold) [reply src/styles/themes/_default-theme.scss:319]: #c97e19

File: src/styles/themes/_default-theme.scss [hunk 10]
```diff
@@ -313,6 +313,10 @@
   --warning-hover-color: #ce811a;
   --warning-hover-background-color: #ffe3b0;
   --warning-hover-border-color: #c97e19;
+  --warning-background1-color: var(--orange-background1-color); // Surface 1
+  --warning-background2-color: var(--orange-background2-color); // Surface 2
+  --warning-background3-color: var(--orange-background3-color); // Surface 3
```
Comment (dkilgore-eightfold) [inline src/styles/themes/_default-theme.scss:318]: Will need UX input on this.

**Review by dkilgore-eightfold — CHANGES_REQUESTED** [review]

File: src/components/Button/button.module.scss (hunk 1 above)
Comment (ykelkar-eightfold) [reply src/components/Button/button.module.scss:1074]: Thanks

File: src/styles/themes/_default-theme.scss (hunk 3 above)
Comment (ykelkar-eightfold) [reply src/styles/themes/_default-theme.scss:873]: Got it, will do

File: src/components/Button/Button.tsx [hunk 11]
```diff
@@ -61,6 +61,7 @@ export const Button: FC<ButtonProps> = React.forwardRef(
       transparent = false,
       type, // TODO: Remove in Octuple v3.0.0 and use `variant` only.
       variant = ButtonVariant.Default,
+      warning = false,
```
[the remark this answers is not in the document]
Comment (ykelkar-eightfold) [reply src/components/Button/Button.tsx:64]: W is after V :)

File: src/components/Button/Button.types.ts (hunk 8 above)
Comment (ykelkar-eightfold) [reply src/components/Button/Button.types.ts:232]: W is after V

File: src/styles/themes/_default-theme.scss (hunk 10 above)
Comment (ykelkar-eightfold) [reply src/styles/themes/_default-theme.scss:318]: Sounds good, I will discuss with David T.

File: src/components/Button/button.module.scss (hunk 1 above)
Comment (ykelkar-eightfold) [reply src/components/Button/button.module.scss]: Fixed it

File: src/styles/themes/_default-theme.scss (hunk 3 above)
Comment (ykelkar-eightfold) [reply src/styles/themes/_default-theme.scss]: Reverted

File: src/styles/themes/_default-theme.scss (hunk 5 above)
Comment (ykelkar-eightfold) [reply src/styles/themes/_default-theme.scss]: Fixed it

File: src/styles/themes/_default-theme.scss (hunk 4 above)
Comment (ykelkar-eightfold) [reply src/styles/themes/_default-theme.scss]: Reverted

File: src/styles/themes/_default-theme.scss (hunk 6 above)
Comment (ykelkar-eightfold) [reply src/styles/themes/_default-theme.scss]: Fixed it

File: src/components/Button/button.module.scss (hunk 9 above)
Comment (ykelkar-eightfold) [reply src/components/Button/button.module.scss]: Reverted

File: src/styles/themes/_default-theme.scss (hunk 10 above)
Comment (ykelkar-eightfold) [reply src/styles/themes/_default-theme.scss:318]: I discussed and showed this to David T. and he approves of these colors as this correlates directly with how disruptive colors use the red color theming:

`--disruptive-backgroundN-color` => `--red-backgroundN-color`
`--warning-backgroundN-color` => `--orange-backgroundN-color`

image
 
I am mimicking this related for warning:
image
image

File: src/components/Button/Button.tsx (hunk 7 above)
Comment (ykelkar-eightfold) [reply src/components/Button/Button.tsx]: Added in UTs 👍

**Comment by ypike-eightfold** [comment]
Hi guys, my question is if the red button (disruptive button) is to make the user stop and think before they click on it to prevent from accidentally deleting, what is the purpose of the orange button for warning? Is it to warn the user before clicking? I think there can be a use case like that but I don't think having a warning button is the right way. 

The alternative approach is like a warning tag on the user interface with a button to fix. Please look a the mockup attached. 

image

**Pull request closed without merging**